File: python_mocra/common/models/priority.py
from enum import Enum, IntEnum
from typing import Protocol, runtime_checkable

class Priority(IntEnum):
    LOW = 1
    NORMAL = 5
    HIGH = 10

    @property
    def suffix(self) -> str:
        if self == Priority.LOW:
            return "low"
        elif self == Priority.NORMAL:
            return "normal"
        elif self == Priority.HIGH:
            return "high"
        return "normal"

@runtime_checkable
class Prioritizable(Protocol):
    def get_priority(self) -> Priority:
        ...

File: src/utils/priority.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize, Default)]
pub enum Priority {
    #[serde(rename = "low")]
    Low = 1,
    #[default]
    #[serde(rename = "normal")]
    Normal = 5,
    #[serde(rename = "high")]
    High = 10,
}

impl Priority {
    pub fn suffix(&self) -> &'static str {
        match self {
            Priority::Low => "low",
            Priority::Normal => "normal",
            Priority::High => "high",
        }
    }
}

pub trait Prioritizable {
    fn get_priority(&self) -> Priority;
}
